time.js

```diff
@@ -1,12 +1,169 @@
-
 require('@js-joda/timezone');
 const time = require('@js-joda/core');
 
 //openHAB uses a RFC DateTime string, js-joda defaults to the ISO version, this defaults RFC instead
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
+
+/**
+ * When formatter is not supplied, uses an RFC DateTimeFormatter that can parse the format
+ * of java.time.ZonedDateTime.toString()
+ * @param {String} text 
+ * @param {time.DateTimeFormatter} formatter 
+ * @returns {time.ZonedDateTime}
+ * @throws {DateTimeParseException} when the text cannot be parsed
+ */
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
     return targetParse(text, formatter);
 }
 
-module.exports = time;
+/**
+ * A passthrough function that can allow users to call `toZonedDateTime` on
+ * almost anything that can be converted to a time.ZonedDateTime without first
+ * checking to see if it's already a time.ZonedDateTime. 
+ * @returns {time.ZonedDateTime} this
+ */
+time.ZonedDateTime.prototype.toZonedDateTime = function () {
```

Comment:
Everywhere I've used this I've done something like the following. Take my [TimerMgr](https://github.com/rkoshak/openhab-rules-tools/blob/main/timerMgr.js). When calling `check` (which looks to see if a timer already exists and creates one if necessary) it accepts a `when` as an argument. That `when` can be:

- `time.ZonedDateTime`
- `String` of the format `PT1h2m3.4s`
- `Number` of milliseconds in all the supported number representations
- `QuantityType<Time>`
- etc.

But in the `TimerMgr`, `LoopingTimer`, `Gatekeeper` and all the rest of the stuff I implement in openhab_rules_tools and my personal code don't need to care. They don't need to do any type checking and iterating through all the possibilities and do something different if it's this type or that type (and all that code becomes duplicated everywhere). All I need is the one line:

    const time = when.toZonedDateTime();

No matter what it is, if it's convertible to a `time.ZonedDateTime` it will be and a meaningful Error thrown when it's not. However, what if it already is a `time.ZonedDateTime`? This function makes it so I don't have to handle that case everywhere. That simple one liner would become more complicated, something like

    const time = (when.toZonedDateTime) ? when.toZonedDateTime() : when; // assume it's a time.ZonedDateTime if there is no `toZonedDateTime()` function

And the error message later on is `when` neither has a `toZonedDateTime()` function nor is a `time.ZonedDateTime` is not as meaningful. But the whole point of all of this PR is to centralize all that sort of type checking and conversion into the library so they don't have to repeat it all over the place.